pkg/auth/template/types.go

```diff
@@ -0,0 +1,24 @@
+package template
+
+import (
+	"github.com/skygeario/skygear-server/pkg/core/config"
+)
+
+const (
+	TemplateItemTypeForgotPasswordEmailTXT           config.TemplateItemType = "forgot_password_email.txt"
```

Comment:
I think these template types should be defined in corresponding feature package instead. This package should be a generic template library and should not have knowledge of individual features.

For example, we can have a `type Template struct { Key string; IsHTML bool; Default string }` and a global function `RegisterTemplate(t Template)`.

Reply:
See if https://github.com/SkygearIO/skygear-server/pull/1087/commits/725fb7495a7ec3dfa98feb1addaa897741a1705e looks good to you

Reply:
I'd prefer unifying `RenderTextTemplate` and `RenderHTMLTemplate` into `RenderTemplate(t T)`, and it will render it as text/HTML according to a flag `IsHTML` in `T`. I've fixed a bug that a HTML template is rendered as text before, so I think we may need this safe guarding here.

Reply:
For some template types, we do not have the corresponding T so I didn't make RenderXXXTemplate to accept T. Also, T has Default, which I think is unnecessary information for RenderXXXTemplate.

Reply:
Okay. So for now we'll accept that we need to be careful when reviewing code so that correct function would be used to render the template.